### ISSUE TITLE
Electrum can't differentiate between corrupt wallets and incorrect password

### ISSUE THREAD

**Issue by AbdussamadA** [spesmilo/electrum#3803]
When full wallet file encryption is used Electrum can't differentiate between an incorrect password and a corrupted wallet. @SomberNight has investigated this and he says the HMAC key is derived from the password so there is no way to tell if the password is wrong or there is a bit flip in the cipher text

@SomberNight  has suggested that a checksum be appended to the cipher text to make to possible to differentiate between the two possibilities.

**Comment by SomberNight** [comment]
Yes, the MAC depends on both the password and the ciphertext, so I can't see an easy way to distinguish corrupted ciphertext and incorrect password.
An extra checksum on the ciphertext stored as plaintext would help detect file corruption, so I think it is something we could consider doing.

**Comment by AbdussamadA** [comment]
Would this be required in the case of partially encrypted wallet files as well? For example if there is a bit flip in the encrypted seed or private keys?
